py/server/deephaven/updateby_clause.py

```diff
@@ -0,0 +1,269 @@
+                default is DECIMAL128.
+
+        Raises:
+            DHError
+        """
+        try:
+            j_builder = _JEmaControl.builder()
+            self.j_ema_control = (j_builder.onNullValue(on_null.value)
+                                  .onNanValue(on_nan.value)
+                                  .onNullTime(on_null_time.value)
+                                  .onNegativeDeltaTime(on_negative_deltatime.value)
+                                  .onZeroDeltaTime(on_zero_deltatime.value)
+                                  .bigValueContext(big_value_context.value).build())
+        except Exception as e:
+            raise DHError(e, "failed to build a EmaControl object.") from e
+
+
+class UpdateByClause(JObjectWrapper):
+    """A UpdateByClause represents an operator for the table update-by operation."""
+
+    j_object_type = _JUpdateByClause
+
+    def __init__(self, j_updateby_clause):
+        self.j_updateby_clause = j_updateby_clause
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_updateby_clause
+
+
+def ema_tick_decay(time_scale_ticks: int, pairs: Union[str, List[str]],
+                   ema_control: EmaControl = None) -> UpdateByClause:
+    """Creates an EMA (exponential moving average) UpdateByClause for the supplied column name pairs, using ticks as
+    the decay unit.
+
+    The formula used is
+        a = e^(-1 / timeScaleTicks)
+        ema_next = a * ema_last + (1 - a) * value
+
+    Args:
+        time_scale_ticks (int): the decay rate (tau) in ticks
+        pairs (Union[str, List[str]]): the input/output column name pairs
+        ema_control (EmaControl): defines how special cases should behave,  when None, the default EmaControl settings
+            as specified in :meth:`~EmaControl.__init__` will be used
+
+    Returns:
+        an UpdateByClause
+
+    Raises:
+        DHError
+    """
+    try:
+        pairs = to_sequence(pairs)
+        if ema_control is None:
+            return UpdateByClause(j_updateby_clause=_JUpdateByClause.Ema(time_scale_ticks, *pairs))
+        else:
+            return UpdateByClause(
+                j_updateby_clause=_JUpdateByClause.Ema(ema_control.j_ema_control, time_scale_ticks, *pairs))
+    except Exception as e:
+        raise DHError(e, "failed to create a tick-decay EMA UpdateByClause.") from e
+
+
+def ema_time_decay(ts_col: str, time_scale_nanos: int, pairs: Union[str, List[str]],
+                   ema_control: EmaControl = None) -> UpdateByClause:
+    """Creates an EMA(exponential moving average) UpdateByClause for the supplied column name pairs, using time as
+    the decay unit.
+
+    The formula used is
+        a = e^(-dt / timeScaleNanos)
+        ema_next = a * ema_last + (1 - a) * value
+
+     Args:
+        ts_col (str): the column in the source table to use for timestamps
+        time_scale_nanos (int): the decay rate (tau) in nanoseconds
```

Comment:
1) `tau` is not listed in the docs.
2) See `time_table`.  The most user friendly use case would be to specify time like '00:00:10', instead of working in nanos.  The methods could support both `int` nanos and `str` values, but we really need string.  If `int` is also supported, the change should also be made to `time_table`.